banks/bank_77.asm

```diff
@@ -1,18 +1,409 @@
+
+	call Func_077_523f
+
+; Intro state manager
+	ld a, [wdcf5]
+	cp 1
+	jr z, .character_slide_up
+	cp 2
+	jr z, .goto_next_screen
+	cp 3
+	jr z, .begin_screen
+	cp 4
+	jr z, .slide_text_up
+
+; slide character right (BG layer)
+	call Func_077_4856
+	jp .Loop
+
+.SkipToTitle
+	ld bc, wcab0
+	xor a
+	ldh [hFFC4], a
+	call Func_096a
+	jp TitleScreen
+
+.character_slide_up
+; Window layer
+	call Func_077_48d9
+	jp .Loop
+
+.slide_text_up
+; Sprite layer
+	call Func_077_48bc
+	jp .Loop
+
+.goto_next_screen
+	ld a, [wdce8]
+	inc a
+	ld [wdce8], a
+	cp 8
+	jr c, .asm_077_41be
+	xor a
+	ld [wdce8], a
+	ld a, 1
+	ld [hFFBF], a
+	ld c, $40
+	call Func_077_5282
+	jp .Loop
+
+.asm_077_41be
+	ld c, $40
+	call Func_077_5282
+	ld a, 0
+	ld [wdcf5], a
+	ld bc, $cab0
+	xor a
+	ldh [hFFC4], a
+	call Func_096a
+	call ClearBGMap0
+	call Func_077_52a6
+	xor a
+	ldh [hFFBF], a
+	ldh [hSCX], a
+	ldh [hFFAF], a
+	ldh [hSCY], a
+	ldh [hFFB1], a
+	ld [wdcf3], a
+	ld [wdcf4], a
+	ld [wdcfb], a
+	ld [wdcf5], a
+	jp .NewScreen
+
+.begin_screen
+	call Func_077_4943
+	ld a, 1
+	ld [wdcf5], a
+	jp .Loop
+
+TitleScreen:
+	call ClearBGMap0
+
+	ld a, BGM_6f
+	call PlaySound
+
+	xor a
+	ldh [hFFBF], a
+	ldh [hSCX], a
+	ldh [hFFAF], a
+	ldh [hSCY], a
+	ldh [hFFB1], a
+	ld [wdce8], a
+	ld [wdcf3], a
+	ld [wdcf4], a
+	ld [wdcfb], a
+	ld [wdcf5], a
+
+	ld hl, vBGMap0
+	ld de, unk_077_6d2f
+	ld bc, $1412
+	ld a, $12
+	ldh [hFF93], a
+	ld a, $14
+	ldh [hFF92], a
+	call PlaceTilemap_Bank0
+
+	ld hl, vBGMap0
+	ld de, unk_077_6bc7
+	ld bc, $1412
+	ld a, $12
+	ldh [hFF93], a
+	ld a, $14
+	ldh [hFF92], a
+	call PlaceAttrmap
+
+	ld hl, unk_077_6b37
+	ld de, $cab0
+	ld bc, $40
+	call CopyBytes3
+
+	ld hl, unk_077_6b7f
+	ld de, $caf0
+	ld bc, $40
+	call CopyBytes3
+
+	ld a, $76
+	ld [wTempBank], a
+	ld hl, $4000
+	ld de, $9000
+	ld bc, $0800
+	call FarCopyBytesVRAM
+
+	ld a, $76
+	ld [wTempBank], a
+	ld hl, $4800
+	ld de, $8800
+	ld bc, $0800
+	call FarCopyBytesVRAM
+
+	ld a, $76
+	ld [wTempBank], a
+	ld hl, $4fa0
+	ld de, $8000
+	ld bc, $0800
+	call FarCopyBytesVRAM
+
+	call Func_077_5289
+	ld a, 8
+	ld [wcd43], a
+	ld a, $10
+	ld [wcd42], a
+	ld a, 1
+	ld [wcd44], a
+	ld a, $81
+	ld [wcd45], a
+	ld a, $40
+	ld [wcd56], a
+	ld a, $68
+	ld [wcd55], a
+	ld a, 2
+	ld [wcd57], a
+	ld a, $82
+	ld [wcd58], a
+	call Func_077_4667
+	ld a, $c7
+	ldh [rLCDC], a
+	ld hl, $cab0
+	xor a
+	ldh [hFFC4], a
+	ldh [hFF9D], a
+	call FadeInPalette
+
+asm_077_42ce:
+	call DelayFrame
+	call Func_077_4667
+	ldh a, [hFF9D]
+	inc a
+	ldh [hFF9D], a
+	ldh a, [hFFBF]
+	and a
+	jr nz, asm_077_4307
+	call Func_077_4525
+	call Func_077_45fa
+	call Func_077_45b4
+	call Func_077_456e
+	ld a, [wdcf5]
+	cp 1
+	jr z, asm_077_42fb
+	cp 2
+	jr z, asm_077_4301
+	call Func_077_4482
+	jp asm_077_42ce
+
+asm_077_42fb:
+	call Func_077_43d6
+	jp asm_077_42ce
+
+asm_077_4301:
+	call Func_077_4359
+	jp asm_077_42ce
+
+asm_077_4307:
+	xor a
+	ldh [hFFBF], a
+	ld [wd0fa], a
+	jp Func_023b
+	ret
+
+Intro_Delay:
+	ld bc, $80 ; ??
```

Comment:
It's clear that they intended to loop this delay for $80 times, but the `cp 0` breaks this.
I think the question marks would be better off next to the `cp 0.`